analysis/Proposed_indicators.md

```diff
@@ -1,6 +1,83 @@
+
+```python
+# Local neighbourhood analysis
+def analyse_local_nh(G, orig_point, buffer=50, length = 1600, intersection_tolerance = 15):
+    # locate closest node on network to 
+    orig_node = ox.get_nearest_node(G, orig_point, return_dist=True)
+    subgraph = nx.ego_graph(G, orig_node[0], radius=length, distance='length')
+    # We project the subgraph to UTM so we can use metres as a unit when buffering
+    subgraph = ox.project_graph(subgraph)
+    su = ox.graph_to_gdfs(subgraph, nodes=False, edges=True, fill_edge_geometry=True)
+    bu = su.geometry.buffer(buffer)
+    uu = bu.geometry.unary_union
+    area_sqm = uu.area
+    area_sqkm = area_sqm*1e-06
+    stats = ox.basic_stats(subgraph, area=area_sqm, clean_intersects=True, tolerance=intersection_tolerance,
+                circuity_dist='euclidean')
+    return({'graph': subgraph,
+             'network_gdfs':su,
+             'buffer':uu,
+             'area_sqkm':area_sqkm,
+             'stats':stats,
+             'origin_node_snap_dist': orig_node[1]})
+     
+# Define some point locations over which we'll iterate
+point_locations = [(-37.808156, 144.964942),(-37.764727, 144.964057)]
+# initialise an 
+nh_estimates = []
+
+for point in point_locations:
+  start = time.time()
+  task = "Calculate local neighbourhood measures for points"
+  nh_estimates.append(analyse_local_nh(G, point))
+  log('Completed task "{}" in {:,.2f} seconds'.format(task,time.time() - start)) 
+  
+# Get some stats
+## area in sqkm
+[x['area_sqkm'] for x in nh_estimates]
+
+## intersection density per sqkm
+[x['stats']['intersection_density_km'] for x in nh_estimates]
+
+## clean intersection density per sqkm
+[x['stats']['clean_intersection_density_km'] for x in nh_estimates]
+```
+So, that makes use of the osmnx stats functionality for calculating street connectivity.  
+
+The use of the function might be overkill; assuming we have a dataframe of sample points, if you coded it right there might even be a vectorised solution to carry out the operations, like extracting the 50m buffered unary union of 1600m network linestrings for a sample point as a geometry field and then use intersection analysis for each sample point's network buffer polygon with other pre-prepared resources:
+
+- use population raster (eg worldpop, or other) to derive a scaled-population density measure to serve as a proxy measure for dwelling density (e.g. take sum of population cells intersecting sausage buffer and divide by some indicative number for population per dwelling (e.g. 4) then standardise that result by dividing by the area of the sausage buffer in Hectares  --- so a result a-bit like dwellings per hectare (although technically, it isn't that.... i'm not saying this is valid, but it could be done and might serve for this purpose)
```

Comment:
In one of my notebook for dwelling density, I use housing density (since we have the data for num. of housing units by block group from the city) as a proxy measure for dwelling density. But I realize this approach may not work for every city due to the data availability. So scaled-population density as you mentioned here would be a better approach to use. Are you using this for the national and Bangkok project as well? I would like to keep this method consistent too.

Reply:
We are not using this approach for our national work; for that we have administrative areas with known official dwelling and person counts.  This approach has its pros and cons though: its official, 'true' data; however, administrative areas have variable sizes (like a street block in the inner city, but potentially much larger in urban fringes) so there is something quite attractive using a regular 100m population grid.  In Bangkok, we are still developing the approach.  We are using the population grid with modelled population estimates, however I have not attempted to derive estimates of dwellings (I think that could be problematic, and might require local knowledge.... so not a priority!  I guess i was thinking out loud to see what others thought about this).